packages/client/hmi-client/src/components/widgets/import-code-button.vue

```diff
@@ -24,60 +35,90 @@
 </template>
 
 <script setup lang="ts">
-import { ref } from 'vue';
+import { ref, computed } from 'vue';
 import Button from 'primevue/button';
 import modal from '@/components/widgets/Modal.vue';
 import { ProjectAssetTypes } from '@/types/Project';
+import { isEmpty } from 'lodash';
 import { getGithubRepositoryContent, getGithubCode } from '@/services/github-import';
 
-defineProps<{
-	url: Object;
+const props = defineProps<{
+	urlString: string;
 }>();
 
-const emit = defineEmits(['open-asset']);
+const emit = defineEmits(['open-code']);
 
-const chosenRepositoryName = ref('');
-const filesToSelect = ref();
+const repositoryName = ref('');
+const currentDirectory = ref('');
+const directoryContents = ref();
 const isModalVisible = ref(false);
 
-async function openCodeBrowser(url) {
+const isInDirectory = computed(() => !isEmpty(currentDirectory.value));
+
+async function initializeCodeBrowser() {
+	currentDirectory.value = ''; // Goes back to root directory if modal is closed then opened again
 	isModalVisible.value = true;
-	chosenRepositoryName.value = url.full_name;
-	filesToSelect.value = await getGithubRepositoryContent(url.contents_url.slice(0, -8));
+	const splitUrl = props.urlString.split('/');
+	repositoryName.value = `${splitUrl[splitUrl.length - 2]}/${splitUrl[splitUrl.length - 1]}`; // owner/repo
```

Comment:
Again to be pedantic, as your code is good as is!
You can use the [Web API URL.pathname](https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname)
```js
repositoryName.value = new URL(props.urlString).pathname.substring(1);
```

Reply:
very cool